src/server/db_slice.cc

```diff
@@ -390,13 +392,27 @@ OpResult<DbSlice::ItAndUpdater> DbSlice::FindMutable(const Context& cntx, string
 }
 
 DbSlice::ItAndExpConst DbSlice::FindReadOnly(const Context& cntx, std::string_view key) {
-  auto res = FindInternal(cntx, key, FindInternalMode::kUpdateReadStats, false);
+  auto res = FindInternal(cntx, key, UpdateStatsMode::kUpdateReadStats,
+                          LoadExternalMode::kDontLoadExternal);
   return {res.it, res.exp_it};
 }
 
 OpResult<PrimeConstIterator> DbSlice::FindReadOnly(const Context& cntx, string_view key,
-                                                   unsigned req_obj_type, bool load_external) {
-  auto it = FindInternal(cntx, key, FindInternalMode::kUpdateReadStats, load_external).it;
+                                                   unsigned req_obj_type) {
+  return FindAReadOnlyInternal(cntx, key, req_obj_type, LoadExternalMode::kDontLoadExternal);
+}
+
+OpResult<PrimeConstIterator> DbSlice::FindAndFetchReadOnly(const Context& cntx,
+                                                           std::string_view key,
+                                                           unsigned req_obj_type) {
+  return FindAReadOnlyInternal(cntx, key, req_obj_type, LoadExternalMode::kLoadExternal);
+}
+
+OpResult<PrimeConstIterator> DbSlice::FindAReadOnlyInternal(const Context& cntx,
+                                                            std::string_view key,
+                                                            unsigned req_obj_type,
+                                                            LoadExternalMode load_mode) {
+  auto it = FindInternal(cntx, key, UpdateStatsMode::kUpdateReadStats, load_mode).it;
```

Comment:
nit, not sure if it's interesting case but if we pass `kLoadExternal` but require a differrent type, we fetch unjustifiably. If we pass optional<req_obj_type> to `FindInternal` too, you will be able to avoid this case.